File: QID-1486-MVAboxcity/MVAboxcity.R

# clear variables and close windows
rm(list = ls(all = TRUE))
graphics.off()

# load data
x  = read.table("cities.txt")
m1 = mean(as.matrix(x))

# Plot box plot
boxplot(x, xlab = "World Cities", ylab = "Values")
lines(c(0.8, 1.2), c(m1, m1), col = "black", lty = "dotted", lwd = 1.2)
title("Boxplot")

# Five Number Summary R 'quantile' function uses a different algorithm to
# calculate the sample quantiles than in the MVA book Therefore, the values
# using Matlab could differ from the Book values, but the difference is not
# great, and should not be significant.  Easiest way to calculate Five Number
# Summary is quantile(population,[.025 .25 .50 .75 .975])
five = quantile(x[, 1], c(0.025, 0.25, 0.5, 0.75, 0.975))

# Display results
print("Five number summary, in millions")
print(five/100)

File: QID-1486-MVAboxcity/MVAboxcity.m
%% Clear variables and close windows
close all
clear
clc

%% Load data
x  = load('cities.txt');
m1 = mean(x);

%% Plot box plot
hold on
boxplot(x,'Symbol','o','label',{'World Cities'},'widths',0.5)
line([0.75 1.25],[m1 m1],'Color','k','LineStyle',':','LineWidth',1.2)
title('Boxplot')
hold off

%% Five Number Summary%
% Matlab "quantile" function uses a different algorithm to calculate the sample quantiles than in the MVA book%
% Therefore, the values using Matlab could differ from the Book values, but
% the difference is not great, and should not be significant.
% Easiest way to calculate Five Number Summary is
% quantile(population,[.025 .25 .50 .75 .975])
five = quantile(x,[.025 .25 .50 .75 .975]);

%% Display results
disp('Five number summary, in millions')
disp('     0.025     0.25      0.5       0.75     0.975 ')
disp(five/100)